From 57f9da244b60954ba27713b3123c17ab1578aacf Mon Sep 17 00:00:00 2001
From: Hannu Lyytinen <hannux@ssrc.tii.ae>
Date: Tue, 28 Feb 2023 02:48:32 +0200
Subject: [PATCH] Use seL4-specific kludge

Signed-off-by: Hannu Lyytinen <hannux@ssrc.tii.ae>
---
 drivers/virtio/virtio_ring.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/virtio/virtio_ring.c b/drivers/virtio/virtio_ring.c
index 603a6f4345ef..452f6eb3e48a 100644
--- a/drivers/virtio/virtio_ring.c
+++ b/drivers/virtio/virtio_ring.c
@@ -260,6 +260,9 @@ static bool vring_use_dma_api(struct virtio_device *vdev)
 	if (xen_domain())
 		return true;
 
+	if (1 /* seL4 */)
+		return true;
+
 	return false;
 }
 
