diff --git a/drivers/input/input.c b/drivers/input/input.c
index 3cfd2c18eebd..873a75814a37 100644
--- a/drivers/input/input.c
+++ b/drivers/input/input.c
@@ -372,8 +372,14 @@ static void input_handle_event(struct input_dev *dev,
 	if (disposition != INPUT_IGNORE_EVENT && type != EV_SYN)
 		add_input_randomness(type, code, value);
 
-	if ((disposition & INPUT_PASS_TO_DEVICE) && dev->event)
-		dev->event(dev, type, code, value);
+	if ((disposition & INPUT_PASS_TO_DEVICE) && dev->event) {
+		/*
+		 * WAR: Disable backpass events to device
+		 * due to the problem with loopback
+		 * between VMs. JIRA: HYPR-440
+		 * dev->event(dev, type, code, value);
+		 */
+	}
 
 	if (!dev->vals)
 		return;
